pkg/graveler/graveler_test.go

```diff
@@ -352,6 +356,144 @@ func TestGraveler_Set(t *testing.T) {
 		})
 	}
 }
+
+func TestGravelerSet_Advanced(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+	newSetVal := graveler.ValueRecord{Key: []byte("key"), Value: &graveler.Value{Data: []byte("newValue"), Identity: []byte("newIdentity")}}
+	// RefManager mock base setup
+	refMgr := mock.NewMockRefManager(ctrl)
+	var getFlow = func() {
+		refMgr.EXPECT().ParseRef(gomock.Any()).Times(1).Return(graveler.RawRef{BaseRef: ""}, nil)
+		refMgr.EXPECT().ResolveRawRef(gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(&graveler.ResolvedRef{
+			Type:                   0,
+			ResolvedBranchModifier: 0,
+			BranchRecord: graveler.BranchRecord{
+				BranchID: "",
+				Branch: &graveler.Branch{
+					CommitID:     "",
+					StagingToken: "",
+					SealedTokens: nil,
+				},
+			},
+		}, nil)
+		refMgr.EXPECT().GetCommit(gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(nil, graveler.ErrCommitNotFound)
+	}
+
+	tests := []struct {
+		name                string
+		expectedValueResult graveler.ValueRecord
+		expectedErr         error
+		committedMgr        *testutil.CommittedFake
```

Comment:
why do we need `committedMgr` if it is new per test - just use one in the test scope

Reply:
Fixed